addons/sale/migrations/12.0.1.1/post-migration.py

```diff
@@ -19,16 +19,82 @@ def type_change_payment_transaction_and_sale_order(env):
         )
 
 
+def fill_sale_order_line_is_expense(cr):
+    # fill is_expense
+    openupgrade.logged_query(
+        cr, """
+        UPDATE sale_order_line sol
+        SET is_expense = TRUE
+        FROM account_analytic_line aal
+        JOIN product_product pp ON aal.product_id = pp.id
+        LEFT JOIN product_template pt ON pp.product_tmpl_id = pt.id
+        WHERE aal.so_line = sol.id AND aal.amount <= 0
+            AND pt.expense_policy IS NOT NULL AND pt.expense_policy != 'no'
```

Comment:
Are you sure this is the good condition? As far as I saw on Friday, it's not the same the line that originates the expense than the extra one that are generated after this. These last ones are the ones which need to be put as `is_expense`, but I fear that with this, you are marking both.

Reply:
I am sure. Check https://github.com/OCA/OpenUpgrade/blob/12.0/addons/sale/models/analytic.py#L22. I tested and works fine.